src/ipc/utils/eigen_ext.tpp

```diff
@@ -84,12 +87,27 @@ project_to_psd(
     }
     Eigen::DiagonalMatrix<double, Eigen::Dynamic> D(eigensolver.eigenvalues());
     // Save a little time and only project the negative values
-    for (int i = 0; i < A.rows(); i++) {
-        if (D.diagonal()[i] < 0.0) {
-            D.diagonal()[i] = 0.0;
-        } else {
-            break;
+    switch (type)
+    {
+    case ProjectType::Clamp:
+    {
+        for (int i = 0; i < A.rows(); i++) {
+            if (D.diagonal()[i] < 0.0) {
+                D.diagonal()[i] = 0.0;
+            } else {
+                break;
+            }
         }
+        break;
+    }
+    case ProjectType::Abs:
+    {
+        for (int i = 0; i < A.rows(); i++)
+            D.diagonal()[i] = std::abs(D.diagonal()[i]);
```

Comment:
The eigen values are sorted, so you can break once you encounter the first non-negative value.

Also, can you use braces here to match the repo style?